### PR TITLE
Enhancement: support `instanceName` filtering for kubernetes auto-discovery deployment

### PR DESCRIPTION
**Pull request by atropos112**
## Proposed change
Add another annotation to kubernetes discovery to allow user to specify for which instance is this supposed to be present for. If no instance provided then it works for all instances preserving current behaviour.

Closes #2487

## Type of change
- [ ] New service widget
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation only
- [ ] Other (please explain)

## Checklist:

- [x] If applicable, I have added corresponding documentation changes.
- [x] If applicable, I have reviewed the [feature](https://gethomepage.dev/latest/more/development/#new-feature-guidelines) and / or [service widget guidelines](https://gethomepage.dev/latest/more/development/#service-widget-guidelines).
- [x] I have checked that all code style checks pass using [pre-commit hooks](https://gethomepage.dev/latest/more/development/#code-formatting-with-pre-commit-hooks) and [linting checks](https://gethomepage.dev/latest/more/development/#code-linting).
- [ ] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.

### CODE REVIEW

**Review by shamoon — COMMENTED** [review]
This certainly looks like it would work but double-checking, did you actually test it?

File: docs/configs/kubernetes.md [hunk 1]
```diff
@@ -93,7 +94,7 @@ spec:
             pathType: Prefix
 ```
 
-When the Kubernetes cluster connection has been properly configured, this service will be automatically discovered and added to your Homepage. **You do not need to specify the `namespace` or `app` values, as they will be automatically inferred.**
+When the Kubernetes cluster connection has been properly configured, this service will be automatically discovered and added to your Homepage. **You do not need to specify the `namespace` or `app` values, as they will be automatically inferred.** In case of multiple instances of homepage, instance annotation can be specified to limit to a specific instance, if no instance is provided, the service will be visible on all instances.
```
Comment (shamoon) [inline docs/configs/kubernetes.md:97]:
```suggestion
When the Kubernetes cluster connection has been properly configured, this service will be automatically discovered and added to your Homepage. **You do not need to specify the `namespace` or `app` values, as they will be automatically inferred.** 

If you are using multiple instances of homepage, an `instance` annotation can be specified to limit services to a specific instance. If no instance is provided, the service will be visible on all instances.
```

**Comment by atropos112** [comment]
You are too quick for me :wink: I have now done testing and can confirm it works. 

My testing was to build the image deploy on my cluster and see 
- having instance annotation matching my instanceName -> I can see it just fine (as expected)
- having instance annotation not matching my instanceName -> I can't see it (as expected)
- not having instance annotation -> I can see it just fine (as expected)

**Comment by shamoon** [comment]
Ha, sorry about that—no rush, but looks good =)

**Review by shamoon — APPROVED** [review]

**Pull request merged**